### DISCUSSION TITLE
Question about `IORING_FEAT_LINKED_FILE`

### DISCUSSION THREAD

**Discussion by realwakka** [axboe/liburing#1312] [Q&A] [answered]
In man page, there is an feature `IORING_FEAT_LINKED_FILE` for `io_uring_setup`
I understand that it enables SQEs could have dependencies like open - read - close. 
Also, in this [Issue](https://github.com/axboe/liburing/issues/1234) I know it can be achived with `io_uring_register_files_sparse()`
It seems that no code in kernel working with `IORING_FEAT_LINKED_FILE`. and I can't find an example for this.
Is it deprecated option? or did I missed something?

**Comment by isilence** [comment, marked as answer]
Note that the flag, just as all other `IORING_FEAT*" flags, is set by the kernel to inform the user space about the behaviour, not the other way around. In other words, you can check it if you need the behaviour, but the user space doesn't control it.

```C
struct io_uring_params params;
io_uring_queue_init_params(..., &params);

if (params.features & IORING_FEAT_LINKED_FILE) {
    // do linked open + read
} else {
    // don't use links
}

```

**Reply by realwakka** [reply]
Thanks for detailed answer. I understood that it's not a option user provides but it shows how to work in kernel.
